spec/unit/hanami/application/settings/dotenv_store_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
```

Comment:
@waiting-for-dev Given this spec messes with the global env (`ENV`), it should be an "isolation" spec. It should go into `spec/isolation`. 🙂

Reply:
I would actually recommend porting this entire spec to `spec/isolation` and removing mocking completely. Dotenv should be treated as an implementation detail and there should be no need to mention it in the specs.

Reply:
Among other things, this file is unit testing the behaviour of of this class when dotenv is or isn't around. I think it's important to have that covered. It's not an implementation detail for sake of this class, it's actually the _whole purpose_ of the class.